docs/grammar.md

```diff
@@ -128,6 +124,51 @@ The name can then be used in the action, like this:
 rule_name[return_type]: '(' a=some_other_rule ')' { a }
 ```
 
+
+### Customizing the generated parser
+
+By default, the generated parser inherits from the Parser class defined in pgen
+Parser, and is named GeneratedParser. One can customize the generated module by
+modifying the header and trailer of the module generated by pegen. To do so one
+can add to the grammar dedicated sections discussed below:
+
+@class NAME
+
+This allows to specify the name of the generated parser.
+
+@header
+
+  Specify the header of the module as a string (one can typically use triple
```

Comment:
The indentation here is not consistent with the other `meta`s. Let's have it consistent everywhere.

Reply:
I went with 2 space which makes for a nicer reading of the source IMO but does not cause the markdown to interprete it as code.